### DISCUSSION TITLE
Third party oAuth providers are returning null?

### DISCUSSION THREAD

**Discussion by jimmiejackson414-zz** [supabase/supabase#2305] [Questions]
I'm trying to implement Facebook, Google and Twitter authentication. So far, I've set up the apps within the respective developer platforms, added those keys/secrets to my Supabase console, and created this graphql resolver:

```
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import camelcaseKeys from 'camelcase-keys';
import { supabase } from 'lib/supabaseClient';
import { LoginInput, Provider } from 'generated/types';
import { Provider as SupabaseProvider } from '@supabase/supabase-js';
import Context from '../../context';
import { User } from '@supabase/supabase-js';

export default async function login(
  _: any,
  { input }: { input: LoginInput },
  { res, req }: Context
): Promise<any> {
  const { provider } = input;

  // base level error object
  const errorObject = {
    __typename: 'AuthError',
  };

  // return error object if no provider is given
  if (!provider) {
    return {
      ...errorObject,
      message: 'Must include provider',
    };
  }

  try {
    const { user, session, error } = await supabase.auth.signIn({
      // provider can be 'github', 'google', 'gitlab', or 'bitbucket'
      provider: 'facebook',
    });
    console.log({ user });
    console.log({ session });
    console.log({ error });

    if (error) {
      return {
        ...errorObject,
        message: error.message,
      };
    }

    const response = camelcaseKeys(user as User, { deep: true });
    return {
      __typename: 'LoginSuccess',
      accessToken: session?.access_token,
      refreshToken: session?.refresh_token,
      ...response,
    };
  } catch (error) {
    return {
      ...errorObject,
      message: error.message,
    };
  }
}
```
I have three console logs set up directly underneath the `signIn()` function, all of which are returning `null`.

I can also go directly to `https://<your-ref>.supabase.co/auth/v1/authorize?provider=<provider>` and auth works correctly, so it appears to have been narrowed down specifically to the `signIn()` function. What would cause the response to return null values?

**Comment by megacherry** [comment]
I have exact the same problem in react native. But in addition i get from `https://<your-ref>.supabase.co/auth/v1/authorize?provider=<provider>`

{"code":400,"msg":"Unsupported provider: Provider is not enabled"}

But the providers are enabled and configured.

**Reply by jimmiejackson414-zz** [reply]
There's now an open issue [here ](https://github.com/supabase/supabase-js/issues/214)for this discussion.
